### PR TITLE
receive_response_message function 

### PR DESCRIPTION
**Pull request by stevenqie**
receive_response_messag-> utilizes the charm ccs CcsRecvResponseMsg function to allow client to receive a message from the server without having to specify the max size

### CODE REVIEW

File: tests/test_server.py [hunk 1]
```diff
@@ -57,3 +62,17 @@ def test_metadata(connected_server):
   assert n_pes == 2
   assert n_nodes == 2
   assert node_size == 1
+
+def test_receive_response_msg(connected_server):
```
Comment (adityapb) [inline tests/test_server.py:66]: Does this need a corresponding Charm++ server code? Could you add that as well to the PR

File: pyccs/server.py [hunk 2]
```diff
@@ -130,6 +136,33 @@ def receive_response(self, max_size: int, timeout: int = DEFAULT_TIMEOUT_PERIOD)
...
+            raise ValueError("Did not receive message/message size is 0")
+        
+        buf_array_type = c_char * retSize.value
+        buf_array = cast(buf, POINTER(buf_array_type)).contents
+
+        # convert back to buffer array 
+        res = bytearray(buf_array[:retSize.value])
+
+        # Free buffer that CcsRecvResponseMsg allocated
+        self._libc.free(buf)
```
Comment (adityapb) [inline pyccs/server.py:162]: Can `res` be accessed still after `buf` is freed?

File: tests/test_server.py (hunk 1 above)
Comment (stevenqie) [reply tests/test_server.py:66]: yes, most of the tests in the file actually need the server to be running.

File: pyccs/server.py (hunk 2 above)
Comment (stevenqie) [reply pyccs/server.py:162]: yes, this should be tested through the test case, but I will write some more extensive ones to make sure

**Comment by stevenqie** [comment]
added bash script so code will start up a server before running the tests + added additional test cases to test the function

**Review by adityapb — APPROVED** [review]
LGTM

**Pull request merged**